mlflow/tracking/fluent.py

```diff
@@ -3054,6 +3055,18 @@ def setup_autologging(module):
 
 
 _ACTIVE_MODEL_ID = ContextVar("active_model_id", default=None)
+_ACTIVE_MODEL_ID_STATE = ContextVar("active_model_id_state", default=None)
+
+
+class ActiveModelIdState(Enum):
+    """
+    Enum to represent the state of the active model ID.
+    It is used to determine whether the active model ID is set by the user or system.
+    If the active model ID is set by the user, MLflow should not modify it.
+    """
+
+    USER = "user"
+    SYSTEM = "system"
```

Comment:
`state` sounds strange because user or system is not a state.

Reply:
/ai Need your help. Suggest different names.

Reply:
Maybe `ActiveModelIdSource`?

Reply:
Can we use a bool (e.g. `set_by_user`)?

Reply:
Yea seems we only need to know if it's set by users or not for now :)